stan/math/prim/mat/err/check_finite.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef STAN_MATH_PRIM_MAT_ERR_CHECK_FINITE_HPP
+#define STAN_MATH_PRIM_MAT_ERR_CHECK_FINITE_HPP
+
+#include <Eigen/Dense>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+
+namespace stan {
+namespace math {
+namespace {
+template <typename T, int R, int C>
+struct finite<Eigen::Matrix<T, R, C>, true> {
+  static void check(const char* function, const char* name,
+                    const Eigen::Matrix<T, R, C>& y) {
+    if (!y.allFinite())
+      domain_error(function, name, y, "is ", ", but must be finite!");
```

Comment:
This degrades the error reporting so that it doesn't say which element.  What does the print look like given that `y` is a matrix?  Does it dump out the whole matrix?

What would you think of having a loop inside the test that picks out which element isn't finite?  We don't care so much how long things take when they fail.